File: backend/app/schemas/event.py
from datetime import datetime
from typing import Optional, Any
from uuid import UUID
from pydantic import BaseModel, Field

class EventBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    start_time: datetime
    end_time: datetime
    location: Optional[str] = None
    category_id: Optional[UUID] = None
    status: str = Field(default="confirmed", pattern="^(confirmed|cancelled|tentative)$")
    metadata: Optional[dict[str, Any]] = {}

class EventCreate(EventBase):
    pass

class EventUpdate(BaseModel):
    title: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    status: Optional[str] = None
    metadata: Optional[dict[str, Any]] = None

class Event(EventBase):
    id: UUID
    user_id: UUID
    ai_generated: bool
    created_at: datetime
    
    class Config:
        from_attributes = True

File: mobile/lib/features/events/data/models/event.dart
class Event {
  final String id;
  final String title;
  final DateTime startTime;
  final DateTime endTime;
  final String? location;
  final String status;
  final String? categoryId;
  final Map<String, dynamic>? metadata;

  Event({
    required this.id,
    required this.title,
    required this.startTime,
    required this.endTime,
    this.location,
    required this.status,
    this.categoryId,
    this.metadata,
  });

  int get importanceScore => (metadata?['importance_score'] as int?) ?? 40;

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as String,
      title: json['title'] as String,
      startTime: DateTime.parse(json['start_time'] as String).toLocal(),
      endTime: DateTime.parse(json['end_time'] as String).toLocal(),
      location: json['location'] as String?,
      status: json['status'] as String? ?? 'confirmed',
      categoryId: json['category_id'] as String?,
      metadata: json['metadata'] as Map<String, dynamic>? ?? {},
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'start_time': startTime.toUtc().toIso8601String(),
      'end_time': endTime.toUtc().toIso8601String(),
      'location': location,
      'status': status,
      'category_id': categoryId,
      'metadata': metadata,
    };
  }
}
